### DISCUSSION TITLE
使用PAGComposition播放多个pag导致其中有几个动画，播放一次会停顿一会儿再播放

### DISCUSSION THREAD

**Discussion by ahsdhajshqwje** [Tencent/libpag#1653] [General] [answered]
const PAG = await window.libpag.PAGInit();

      let id = this.renderingId

      let pagComposition = pagComposition = PAG.PAGComposition.make( 3200, 1120 );

// 这里的urlArray 则是我的pag动画
       for (let i = 0; i < this.urlArray.length; i++) {

          let url = "/pag/" + this.urlArray[i] + ".pag"

          const buffer = await fetch(url).then((response) => {
            return response.arrayBuffer()
          })

          const pagFile = await PAG.PAGFile.load(buffer)

          pagComposition.addLayer(pagFile);
        }
      }
      const canvas = document.getElementById(id);
      if (canvas !== null) {
        canvas.width = pagComposition.width();
        canvas.height = pagComposition.height();

        const pagView = await PAG.PAGView.init(pagComposition, canvas);

        pagView.setRepeatCount(0);

        await pagView.play();

        this.canvasRef = pagView
      }

**Comment by domchen** [comment, marked as answer]
PAGComposition组合的是一个带固定时间轴的预合成，里面的每个PAGFile都是放在它总的时间轴上，如果子PAGFile长度不一样，就会停在最后一帧，等总的PAGComposition时间轴都播放结束再从头开始。这个组合并不是一个没时间轴的让子PAGFile各自分别按自己时间轴无限循环播放逻辑。你最简单的方式是把几个PAG文件的时长都做成一样的就行了。
